Toggl.Storage.Realm/Queries/ProcessPullResultQuery.cs

```
@@ -230,7 +227,8 @@ private void updateTimeEntry(ITimeEntry timeEntry, RealmTimeEntry dbTimeEntry, R
 
             dbTimeEntry.RealmTags.Clear();
             tagsIds
-                .Select(tagId => realm.Find<RealmTag>(tagId))
+                .Select(tagId => realm.GetById<RealmTag>(tagId))
+                .Where(tag => tag != null)
```

Shouldn't this fail if there are no tags, rather than just silently being ignored?

---

Or is this also the case of potentially locally deleted thing after the push?

---

I am not sure which one is better. I noticed that in `RealmConstructors.cs` we would throw (we use `Single` to fetch the entity by ID) if the entity was not there. I'll remove the `Where`.